pkg/controllers/managementuserlegacy/helm/common/common.go

```diff
@@ -215,8 +216,12 @@ func GenerateAnswerSetValues(app *v3.App, tempDir *HelmPath, extraArgs map[strin
 	// a user-supplied values file will overridden default values.yaml
 	if app.Spec.ValuesYaml != "" {
 		custom := "custom-values.yaml"
-		valuesYaml := filepath.Join(tempDir.FullPath, custom)
-		if err := ioutil.WriteFile(valuesYaml, []byte(app.Spec.ValuesYaml), 0755); err != nil {
+		valuesYamlFileName := filepath.Join(tempDir.FullPath, custom)
+		valuesYaml, err := compression.DecompressValuesYaml(app.Spec.ValuesYaml)
```

Comment:
The only concern I have is to make sure that `compression.DecompressValuesYaml` should work against both compressed and uncompressed content, as it will break for exiting app that uses uncompressed values.yaml. I guess it is already the case since you check for magic header, just to double check.

Reply:
Or are we migrating existing app.Spec.ValuesYaml into compressed content?

Reply:
I spoke with @dramich, and we want to use `string` instead of `[]byte` for the values. That was an intentional change since the last time we looked it this.

Since we are using the same field, we need to check if it's actually compressed with GZIP, as you mentioned. That's what the byte slice check is for.

I believe the current code covers that scenario. Do you see any room for improvement?